crates/sui-indexer-alt-framework/src/handlers/cp_mapping.rs

```diff
@@ -20,8 +20,61 @@ pub struct StoredCpMapping {
     pub epoch: i64,
 }
 
+pub struct PrunableRange {
+    from: StoredCpMapping,
+    to: StoredCpMapping,
+}
+
 pub struct CpMapping;
 
+impl PrunableRange {
+    /// Gets the tx and epoch mappings for both the start and end checkpoints.
+    ///
+    /// The values are expected to exist since the cp_mapping table must have enough information to
+    /// encompass the retention of other tables.
+    pub async fn get_range(
+        conn: &mut Connection<'_>,
+        from_cp: u64,
+        to_cp: u64,
+    ) -> QueryResult<PrunableRange> {
+        let results = cp_mapping::table
+            .select(StoredCpMapping::as_select())
+            .filter(cp_mapping::cp_sequence_number.eq_any([from_cp as i64, to_cp as i64]))
+            .order(cp_mapping::cp_sequence_number.asc())
+            .load::<StoredCpMapping>(conn)
+            .await?;
+
+        match results.as_slice() {
+            [first, .., last] => Ok(PrunableRange {
+                from: first.clone(),
+                to: last.clone(),
+            }),
+            _ => Err(diesel::result::Error::NotFound),
+        }
+    }
+
+    /// Inclusive start and exclusive end range of prunable checkpoints.
+    pub fn checkpoint_interval(&self) -> (u64, u64) {
+        (
+            self.from.cp_sequence_number as u64,
+            self.to.cp_sequence_number as u64,
+        )
+    }
+
+    /// Inclusive start and exclusive end range of prunable txs.
+    pub fn tx_interval(&self) -> (u64, u64) {
+        (self.from.tx_lo as u64, self.to.tx_lo as u64)
+    }
+
+    /// Returns the epochs that contain the checkpoints in this range.
+    ///
+    /// While the checkpoint and tx ranges use exclusive end bounds, the epoch is different in that
+    /// it represents which epoch the `from` and `to` checkpoints come from.
```

Comment:
I would rather add 1 to the `to` epoch to make it consistent than explain the difference here.